lib/gltfPipeline.js

```diff
@@ -117,41 +116,52 @@ function processFile(inputPath, options, callback) {
     });
 }
 
-function writeFile(gltf, outputPath, options, callback) {
+function writeFile(gltf, outputPath, options) {
     var fileExtension = path.extname(outputPath);
     var binary = defaultValue(options.binary, false);
     var embed = defaultValue(options.embed, true);
     var embedImage = defaultValue(options.embedImage, true);
     var createDirectory = defaultValue(options.createDirectory, true);
-    if (binary || fileExtension === '.glb') {
-        writeBinaryGltf(gltf, {
-            outputPath : outputPath,
-            embed : embed,
-            embedImage : embedImage,
-            createDirectory : createDirectory
-        }, callback);
-    } else {
-        writeGltf(gltf, outputPath, embed, embedImage, createDirectory, callback);
-    }
+
+    return new Promise(function() {
```

Comment:
I don't quite understand when `Promise`s should have a `resolve` parameter. 

I previously had this function take a `resolve` and called `resolve()` at the end of the function, but this seemed to caused `writeFile(...).then` to be triggered preemptively. 

Do promises that ultimately return other promises not need `resolve` (since the last promise on the chain will trigger the resolve?)